src/test/py/bazel/bzlmod/bazel_lockfile_test.py

```diff
@@ -299,6 +299,55 @@ def testModuleExtension(self):
     )
     self.assertNotIn('Hello from the other side!', ''.join(stderr))
 
+  def testIsolatedModuleExtension(self):
```

Comment:
This is exactly the testModuleExtention function. Maybe it's better to just add "isolate = True" there, and test here that changing it results in a new module extension added to the lockfile?